### PR TITLE
Automated cherry pick of #85115: remove disk locks per vm

### PR DESCRIPTION
**Pull request by aramase**
Cherry pick of #85115 on release-1.16.

#85115: remove disk locks per vm

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

### CODE REVIEW

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_controller_common.go [hunk 1]
```diff
@@ -170,16 +171,20 @@ func (c *controllerCommon) DetachDisk(diskName, diskURI string, nodeName types.N
 	klog.V(2).Infof("detach %v from node %q", diskURI, nodeName)
 
 	// make the lock here as small as possible
-	diskOpMutex.LockKey(instanceid)
+	c.vmLockMap.LockEntry(strings.ToLower(string(nodeName)))
+	c.diskAttachDetachMap.Store(strings.ToLower(diskURI), "detaching")
```
Comment (andyzhangx) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_controller_common.go]: this should be in another cherry pick

Comment (aramase) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_controller_common.go]: done!

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**